src/ex_docmd.c

```diff
@@ -8928,6 +8928,10 @@ ex_cd(exarg_T *eap)
 	    /* Echo the new current directory if the command was typed. */
 	    if (KeyTyped || p_verbose >= 5)
 		ex_pwd(eap);
+
+#ifdef FEAT_AUTOCMD
+	    apply_autocmds(EVENT_DIRCHANGED, new_dir, new_dir, FALSE, curbuf);
```

Comment:
> I find it more consistent with the other autocommands if it is a filename glob.

The pattern is not always a filename (e.g. `FileType` autocommand), and it would be useless for this use-case. 

The autocmd pattern is the primary way of selecting a subset of event messages to listen to.